server_test.go

```diff
@@ -197,13 +216,26 @@ func TestServing(t *testing.T) {
 	if txt != "Hello example" {
 		t.Error("unexpected result for example.com", txt, "!= Hello example")
 	}
+
+	// Test TLS State. With UDP, it should be OFF.
+	m.SetQuestion("tlsstate.example.net.", TypeTXT)
+	r, _, err = c.Exchange(m, addrstr)
+	if err != nil {
+		t.Error("failed to exchange eXaMplE.cOm", err)
+	}
+	txt = r.Extra[0].(*TXT).Txt[0]
+	if txt != TLSOffMsg {
+		t.Error("unexpected result for tlsstate.examepl.net.", txt, "!= "+TLSOffMsg)
+	}
 }
 
 func TestServingTLS(t *testing.T) {
 	HandleFunc("miek.nl.", HelloServer)
 	HandleFunc("example.com.", AnotherHelloServer)
+	HandleFunc("tlsstate.example.net.", TLSStateServer)
```

Comment:
Adding your test to this testcase is a bit unfortunate and inflates the PR also. If possible can you just make a small TLS test server that gets 2 reqs. UDP and TCP+TLS and see if you can access in the connection state *in* the handler ,so you don't need to do this TXT record dance?

Reply:
ok, will fix this.